src/include/platform/internal/GenericDeviceInstanceInfoProvider.ipp

```diff
@@ -21,6 +21,20 @@ namespace chip {
 namespace DeviceLayer {
 namespace Internal {
 
+template <class ConfigClass>
+CHIP_ERROR GenericDeviceInstanceInfoProvider<ConfigClass>::GetDeviceName(MutableCharSpan & deviceNameSpan)
+{
+
+    constexpr char deviceName[] = CHIP_DEVICE_CONFIG_DEVICE_NAME;
+
+    ReturnErrorCodeIf(sizeof(deviceName) > deviceNameSpan.size(), CHIP_ERROR_BUFFER_TOO_SMALL);
+    ReturnErrorCodeIf(sizeof(deviceName) != ConfigurationManager::kMaxDeviceNameLength, CHIP_ERROR_BUFFER_TOO_SMALL);
+    memcpy(deviceNameSpan.data(), deviceName, sizeof(deviceName));
+    deviceNameSpan.reduce_size(sizeof(deviceName));
```

Comment:
This is wrong: it's including the null terminator in the size, which is not how CharSpan is supposed to work.  It should contain only the actual string data.

Reply:
updated